pkg/service/client_grpc.go

```diff
@@ -101,39 +99,31 @@ func NewGRPCClient(conn *grpc.ClientConn, logger log.Logger) KolideService {
 
 // dialGRPC creates a grpc client connection.
 func DialGRPC(
-	serverURL string,
-	insecureTLS bool,
-	insecureTransport bool,
-	certPins [][]byte,
+	k types.Knapsack,
 	rootPool *x509.CertPool,
-	logger log.Logger,
 	opts ...grpc.DialOption, // Used for overrides in testing
 ) (*grpc.ClientConn, error) {
-	level.Info(logger).Log(
-		"msg", "dialing grpc server",
-		"server", serverURL,
-		"tls_secure", insecureTLS == false,
-		"transport_secure", insecureTransport == false,
-		"cert_pinning", len(certPins) > 0,
+
+	k.Slogger().Debug("dialing grpc server",
+		"server", k.KolideServerURL(),
+		"tls_secure", k.InsecureTLS() == false,
+		"transport_secure", k.InsecureTransportTLS() == false,
+		"cert_pinning", len(k.CertPins()) > 0,
 	)
+
 	grpcOpts := []grpc.DialOption{
 		grpc.WithTimeout(time.Second),
 	}
-	if insecureTransport {
+	if k.InsecureTransportTLS() {
 		grpcOpts = append(grpcOpts, grpc.WithInsecure())
 	} else {
-		host, _, err := net.SplitHostPort(serverURL)
-		if err != nil {
-			return nil, fmt.Errorf("split grpc server host and port: %s: %w", serverURL, err)
-		}
-
-		creds := &tlsCreds{credentials.NewTLS(makeTLSConfig(host, insecureTLS, certPins, rootPool, logger))}
+		creds := &tlsCreds{credentials.NewTLS(makeTLSConfig(k, rootPool))}
```

Comment:
so the old code was splitting the host form port in the server url, is that still needed? will test against local k2 with port

Reply:
yes, this is a thing ... working on it

Reply:
now extracting port inside makeTlsConfig